src/streamlink/session.py

```diff
@@ -387,13 +392,40 @@ def resolve_url(self, url, follow_redirect=True):
         :param follow_redirect: follow redirects
 
         """
+        default_plugin_url_copy = url
+        default_plugin = self.get_option("default-plugin")
         url = update_scheme("http://", url)
 
         available_plugins = []
+
         for name, plugin in self.plugins.items():
             if plugin.can_handle_url(url):
                 available_plugins.append(plugin)
 
+        if len(available_plugins) == 0 and default_plugin is not None:
+            plugin_found = False
+
+            for name, plugin in self.plugins.items():
+                if plugin.__name__.lower() == default_plugin.lower():
+                    plugin_found = True
+                    for string in plugin.get_urls():
+                        if plugin.can_handle_url(string + default_plugin_url_copy):
+                            try:
+                                plugininstance = plugin(string + default_plugin_url_copy)
+                                streams = plugininstance.streams(None, None)
+                                available_plugins.append(plugin)
+                                url = string + default_plugin_url_copy
+                                break
+                            except (NoPluginError, PluginError) as e:
+                                pass
+
+                    if len(available_plugins) == 0:
+                        print("No streams with default plugin " + default_plugin + " found, plugin either doesnt support url shortening or url is not valid. Try without --default-plugin.")
```

Comment:
Please use the appropriate logging methods and watch the max line length.
https://github.com/streamlink/streamlink/blob/d022e9b8fde1570727267ff0e39c5cf6f6c87e2c/src/streamlink/session.py#L22

Is it even necessary to log something here? I think a general log message which says that it is looking for a fallback plugin now when the parameter is set and the `available_plugins` is empty is more fitting. The message below could then also be removed, as it is unnecessary as well.